jinahub/indexers/searcher/AnnoySearcher/__init__.py

```diff
@@ -84,7 +87,18 @@ def search(self, docs: DocumentArray, parameters: Dict, **kwargs):
             )
             for idx, dist in zip(indices, dists):
                 match = Document(id=self._ids[idx], embedding=self._vecs[idx])
-                match.scores['distance'] = 1 / (1 + dist)
+                if self.is_distance:
+                    if self.metric == 'dot':
+                        match.scores[self.metric] = 1 - dist
+                    else:
+                        match.scores[self.metric] = dist
+                else:
+                    if self.metric == 'angular' or self.metric == 'hamming':
+                        match.scores[self.metric] = 1 - dist
+                    elif self.metric == 'euclidean' or self.metric == 'manhattan':
+                        match.scores[self.metric] = 1 / (1 + dist)
+                    else:
+                        match.scores[self.metric] = dist
```

Comment:
it is weird to say `is_distance` False and set `score` to `distance`

Reply:
This line is only for `dot`, the inner product metric. The annoy already output `dot` as a similarity score.
So the `is_distance ==True` leads to `1-dist`, `is_distance ==False` leads to original `dist`.
Should I add some comments above these code?

Reply:
I refactored those check